lib/created_message_queue_handler.ts

```diff
@@ -258,77 +287,58 @@ export async function handleMessage(
   return right(errorOrNotification.value);
 }
 
-export function processGenericError(error: Error): void {
-  // TODO: update message status (FAILED)
-  winston.error(`CreatedMessageQueueHandler|Unexpected error|${error.message}`);
-}
-
 /**
  * Returns left(true) in case the caller must
  * trigger a retry calling context.done(true)
  */
 export async function processRuntimeError(
-  queueService: QueueService,
-  error: RuntimeError,
-  queueMessage: IQueueMessage
+  lQueueService: QueueService,
+  messageStatusUpdater: MessageStatusUpdater,
+  queueMessage: IQueueMessage,
+  error: RuntimeError
 ): Promise<Either<boolean, Option<OutputBindings>>> {
   if (isTransient(error)) {
     winston.warn(`CreatedMessageQueueHandler|Transient error|${error.message}`);
     const shouldTriggerARetry = await updateMessageVisibilityTimeout(
-      queueService,
+      lQueueService,
       MESSAGE_QUEUE_NAME,
       queueMessage
     );
+    const errorOrUpdatedMessageStatus = await messageStatusUpdater(
+      MessageStatusValueEnum.THROTTLED
+    );
+    if (isLeft(errorOrUpdatedMessageStatus)) {
+      winston.warn(
+        `CreatedMessageQueueHandler|Transient error|${
+          errorOrUpdatedMessageStatus.value.message
+        }`
+      );
+    }
     return left(shouldTriggerARetry);
   } else {
-    // TODO: update message status (FAILED)
     winston.error(
       `CreatedMessageQueueHandler|Permanent error|${error.message}`
     );
-    // return no bindings so message processing stops here
-    return right(none);
+    const errorOrUpdatedMessageStatus = await messageStatusUpdater(
+      MessageStatusValueEnum.FAILED
+    );
+    if (isLeft(errorOrUpdatedMessageStatus)) {
+      // if we get an error updating the message status
+      // we retry the whole procedure (it will log the error anyway)
+      const transientError = errorOrUpdatedMessageStatus.value;
+      return await processRuntimeError(
```

Comment:
this is a recursive call which "translates" a permanent error to a transient error (the status update failed, we trigger a retry)